FreeRTOS-Plus/Demo/FreeRTOS_Plus_POSIX_with_actor_Windows_Simulator/lib/FreeRTOS-Plus-POSIX/source/FreeRTOS_POSIX_pthread.c

```diff
@@ -409,11 +409,19 @@ int pthread_join( pthread_t pthread,
     int iStatus = 0;
     pthread_internal_t * pxThread = ( pthread_internal_t * ) pthread;
 
+    if (pxThread == NULL) 
+    {
+        iStatus = ESRCH;
```

Comment:
In most of our POSIX API implementation, I believe we only implement "required" error codes, not "optional" or "recommended" error codes. 

I'm voting down this change, since -- 
1. We need to strictly stick to specification. https://pubs.opengroup.org/onlinepubs/9699919799/functions/pthread_join.html
```ESRCH``` is for 

> If an implementation detects use of a thread ID after the end of its lifetime, it is recommended that the function should fail and report an [ESRCH] error.

Invalid thread handle is not equivalent to "after the end of its lifetime". 

2. Our implementation does not support error code ```ESRCH```.
https://github.com/FreeRTOS/Lab-Project-FreeRTOS-POSIX/blob/master/include/FreeRTOS_POSIX/errno.h. 
This code shouldn't compile. 

Unless you have better reason to add this...

Reply:
Hi, 
Thanks for taking the time to review this, I noticed the list of reviewers to choose from was sparse for this repo. Some background, I reused the visual studio project included in the repo for building and ran this over windows sim.

I agree that ESCRH should be avoided since we don't officially support and confused why my code compiled without issue despite 2.)

I disagree with 1.) and stand by returning ESRCH as valid reaction for null thread handles. Several POSIX implementations, including gllibc used by Linux, do so. My laptop also returns ESRCH when I pthread_join(NULL, ...).

Reply:
I'll remove ESCRH from this PR. Makes sense. If some functions handle it, all appropriate functions should handle it and that's not the point of this PR

Reply:
Regarding to 1, there are a variety of implementations to POSIX standard. Everyone claims their implementation compiles with standard, while the reality is every implementation is slightly different. When our code placed with vendor code, these subtle differences always cause trouble. (Unless the argument here is about how the standard is interpreted -- to be specific, whether "invalid thread handle is or not equivalent to "after the end of its lifetime"." )

So, no, please stick to "The Open Group Base Specifications Issue 7, 2018 edition IEEE Std 1003.1-2017 (Revision of IEEE Std 1003.1-2008)".